plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -204,7 +208,9 @@ public class IcebergPageSourceProvider
     private final OrcReaderOptions orcReaderOptions;
     private final ParquetReaderOptions parquetReaderOptions;
     private final TypeManager typeManager;
-    private final DeleteManager deleteManager;
+    private final DeleteManager unpartitionedTableDeleteManager;
+    private final Map<Integer, Function<PartitionData, PartitionKey>> partitionKeyFactories = new ConcurrentHashMap<>();
```

Comment:
This came from the orginal code, but seems off to me.  The map key is `partitionSpec.specId()`, but since the page source will be shared across multiple tables, I expect this will need the table name also (unless there is a better identifier in Iceberg).

Reply:
The page source provider seems to be created on a per-table-scan basis. For example, it's created in `ScanFilterAndProjectOperatorFactory`. Am I misunderstanding?